crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -702,6 +705,39 @@ where
                 .ok_or(Error::<Block>::MissingSalt(hash))?
                 .salt;
 
+            let parent_block_id = BlockId::Hash(*block.header.parent_hash());
+
+            let max_plot_size = self
+                .client
+                .runtime_api()
+                .max_plot_size(&parent_block_id)
+                .expect("Failed to get `max_plot_size` from runtime API");
+            let total_number_of_pieces = {
+                let merkle_num_leaves = {
+                    let record_size = self
+                        .client
+                        .runtime_api()
+                        .record_size(&parent_block_id)
+                        .expect("Failed to get `record_size` from runtime API");
+                    let recorded_history_segment_size = self
+                        .client
+                        .runtime_api()
+                        .recorded_history_segment_size(&parent_block_id)
+                        .expect("Failed to get `recorded_history_segment_size` from runtime API");
+
+                    u64::from(recorded_history_segment_size / record_size * 2)
+                };
+
+                let total_number_of_segments = self
```

Comment:
Let's return number of pieces this runtime API call instead. It will remove 2 unnecessary runtime calls above and might potentially be useful in the future if we decide to make segments variable in size. Just make sure to add TODO like https://github.com/subspace/subspace/blob/d8e2877376aa3b502a96eaebf2981e2d1c8e4526/crates/sc-consensus-subspace/src/lib.rs#L904 there before we made the final decision (it is likely that segments will remain constant size, but not 100% guaranteed yet).